Library/ViewModels/PledgeStatusLabelViewModelTests.swift

```diff
@@ -0,0 +1,240 @@
+      let creator = User.template
+        |> User.lens.id .~ 5
+
+      let project = Project.cosmicSurgery
+        |> Project.lens.creator .~ creator
+        |> Project.lens.state .~ .successful
+        |> Project.lens.personalization.backing .~ (
+          Backing.template
+            |> Backing.lens.status .~ .errored
+        )
+
+      self.vm.inputs.configure(with: project)
+      self.vm.inputs.viewDidLoad()
+
+      self.labelTextString.assertValues([
+        "We can’t process your pledge. Please update your payment method."
+      ])
+    }
+  }
+
+  func testBackingStatus_Pledged() {
+    withEnvironment(currentUser: .template) {
+      let creator = User.template
+        |> User.lens.id .~ 5
+
+      let project = Project.cosmicSurgery
+        |> Project.lens.creator .~ creator
+        |> Project.lens.state .~ .successful
+        |> Project.lens.stats.currentCurrency .~ Currency.USD.rawValue
+        |> Project.lens.stats.currency .~ Currency.USD.rawValue
+        |> Project.lens.country .~ .us
+        |> Project.lens.personalization.backing .~ (
+          Backing.template
+            |> Backing.lens.status .~ .pledged
+        )
+
+      self.vm.inputs.configure(with: project)
+      self.vm.inputs.viewDidLoad()
+
+      self.labelTextString.assertValues([
+        "If the project reaches its funding goal, you will be charged on October 16, 2016."
+      ])
+    }
+  }
+
+  func testBackingStatus_Pledged_OtherCurrency() {
+    withEnvironment(currentUser: .template, locale: Locale(identifier: "en")) {
+      let creator = User.template
+        |> User.lens.id .~ 5
+
+      let project = Project.cosmicSurgery
+        |> Project.lens.creator .~ creator
+        |> Project.lens.stats.currentCurrency .~ Currency.USD.rawValue
+        |> Project.lens.stats.currency .~ Currency.HKD.rawValue
+        |> Project.lens.country .~ .hk
+        |> Project.lens.state .~ .successful
+        |> Project.lens.personalization.backing .~ (
+          Backing.template
+            |> Backing.lens.status .~ .pledged
+        )
+
+      self.vm.inputs.configure(with: project)
+      self.vm.inputs.viewDidLoad()
+
+      self.labelTextString.assertValues([
+        "If the project reaches its funding goal, you will be charged HK$ 10 on October 16, 2016."
+      ])
+    }
+  }
+
+  func testBackingStatus_AllOtherStatuses() {
+    withEnvironment(currentUser: .template) {
+      let statuses = Backing.Status.allCases
+        .filter { ![.canceled, .collected, .dropped, .errored, .pledged].contains($0) }
+
+      statuses.forEach {
+        let creator = User.template
+          |> User.lens.id .~ 5
+
+        let project = Project.cosmicSurgery
+          |> Project.lens.creator .~ creator
+          |> Project.lens.state .~ .successful
+          |> Project.lens.personalization.backing .~ (
+            Backing.template
+              |> Backing.lens.status .~ $0
+          )
+
+        self.vm.inputs.configure(with: project)
```

Comment:
Do you need to call `self.vm.inputs.viewDidLoad()` somewhere?

Reply:
😬 😬